spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -159,19 +164,36 @@
 
 	/**
 	 * Map of serviceId's -> tag to query for in server list. This allows filtering
-	 * services by a single tag.
+	 * services by a single tag. (Deprecated - instead use 'serviceInstanceQueryTags')
 	 */
+	@Deprecated
 	private Map<String, String> serverListQueryTags = new HashMap<>();
 
+	/**
+	 * Tag to query for in service list if one is not listed in serverListQueryTags.
+	 * (Deprecated - instead use 'defaultServiceInstanceQueryTags')
+	 */
+	@Deprecated
+	private String defaultQueryTag;
+
+	/**
+	 * Map of serviceId -> tags to use when querying for service instances. This allows
+	 * filtering service instances by tags.
+	 */
+	private Map<String, List<String>> serviceInstanceQueryTags = new HashMap<>();
```

Comment:
This pair is the new replacement - named accordingly for what it is being used for (as well as I had to deal w/ the close names of the deprecated pair).